porting/configure_test_fix/Overlay.rst

```diff
@@ -3,82 +3,133 @@
 
-When rebuilding the system image after small changes like these, you need not ``mka clean`` first. However, changes to PRODUCT_PROPERTY_OVERRIDES might not get detected by the build system. To ensure this, go to the directory ``BUILDDIR/out/target/product/CODENAME/system`` and delete the file ``build.prop`` before rebuilding.
+After a successful build, the resulting ``system.img`` must reflashed together with the UBports root filesystem. On boot files and – in case of Ubuntu Touch 20.04 or later – marked directories in ``/android/system/halium`` will be mounted onto their final locations which are derived from their path relative to the designated base directory.
 
 Overlay method for Gitlab CI script-based builds
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Builds in this category can also be adapted to a particular device using overlay files, and when building by this method (see :ref:`Gitlab-CI`) it is possible not only to replace existing files, but also to introduce new ones.
-
-When you have prepared the files to replace or introduce, and determined the exact locations where they need to go on your device, these locations need to be replicated under the directory ``overlay/system/`` before running the Gitlab CI scripts according to `the instructions included. <https://gitlab.com/ubports/community-ports/android9/xiaomi-redmi-note-7-pro/xiaomi-violet/-/blob/master/README.md>`_
+When using Gitlab CI script-based builds (see :ref:`Gitlab-CI`) overlay files and – in case of Ubuntu Touch 20.04 and later – overlay directories have to be placed in the repository under ``overlay/system``.  Absolute destination paths are derived based on the location of a directory or file relative to the overlay directory tree.  Note that the files or directories under the destination path must neither be writable nor mount points themselves. In case of Ubuntu Touch 16.04 new files and directories can only be introduced by directly copying them to their final destination. Ubuntu Touch 20.04 or later versions should always use overlay or override directories as explained above.
 
-For example:
-""""""""""""
+Example
+"""""""
 
-The udev rules file ``70-android.rules`` needs to go into ``system/halium/lib/udev/rules.d/``. This is accomplished by first creating the directory in the Gitlab scripts build tree. Standing in the root of this tree, first create the directory and then copy the file to this location::
+Following the above example, adding the overlay files ``/lib/udev/rules.d/70-android.rules`` and ``/etc/ubuntu-touch-session.d/android.conf`` requires them to end up below ``/opt/halium-overlay`` on the resulting filesystem, that is as ``/opt/halium-overlay/lib/udev/rules.d/70-android.rules`` and ``/opt/halium-overlay/etc/ubuntu-touch-session.d/android.conf``. This is achieved by placing them as ``overlay/system/lib/udev/rules.d/70-android.rules`` and ``overlay/system/etc/ubuntu-touch-session.d/android.conf`` in the repository.
```

Comment:
> This is achieved by placing them as ``overlay/system/lib/udev/rules.d/70-android.rules``...

As I discussed earlier, this is incorrect. At least without a modification to the build script...

I just realised that, to use mount-based overlay, one has to actually do "directly extracted" overlay to `/opt/halium-overlay` first. How can we eliminate this confusion...

Like, as it currently stands, porters can choose to use "directly extracted" overlay by placing files directly on `overlay/system`, or choose to use "mount-based overlay" by "directly extracted" overlay over `/opt/halium-overlay` i.e. placing files at `overlay/system/opt/halium-overlay`.

Reply:
Corrected for the example. Do you think there we need a second example showing directly extracted files for 16.04?

Reply:
The second example is added in the new version.